### ISSUE TITLE
Id property is not loaded in projected entity when Relay is enabled and the Relay Id is built using another property

### ISSUE THREAD

**Issue by ignaciojocano-ITH** [ChilliCream/hotchocolate#4041]
**Describe the bug**
While having the option `EnableRelaySupport()`, and then having a class `Wrapper` which enables accessing another class `SubClass`, if the `SubClass` has a custom string to represent a composite key, the original `Id` column from `SubClass` cannot be used to generate such string.
The `Id` property from `SubClass` is not loaded through the database query performed by HC.

**To Reproduce**
You need to 
- Have two entities A and B related, and B needs to have a property calculated with the B.Id property.
- Have EnableRelaySupport()
- Have a Type for B that implements the Relay node with an IdField that uses a calculated property. And the calculated property uses the original Id column of B

Having the previous setup
1. Perform a query to retrieve A and the related class B including the Id
2. Inspect the hashed Id returned for B
3. See that the hashed string was built using a non initialized Id to generate the custom composed key

**Expected behavior**
The entire B entity should be loaded from the database, including the Id property (which in this case is used to generate a composed key)

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Version 11.3.2

**Additional context**
I wrote a rough implementation to share the issue here https://github.com/ignj/hc-composedkey/tree/composedkey_id_not_loaded
Basically, performing the query
```graphql
query {
  wrappers {
    example
    {
      id
      typeId
    }
    id
    typeId
    exampleId
  }
}
```
returns the hashed `id` for `example` `RXhhbXBsZQpkeyJJZCI6MCwiVHlwZUlkIjo0fQ==` which is 
```
Example
d{"Id":0,"TypeId":4}
```
but the only existent entry does not have the Id = 0

By setting a breakpoint in https://github.com/ignj/hc-composedkey/blob/e0c81edd7cbf6c3ff0c87c469e9e517fa5dd12d1/ApplicationDbContext.cs#L15 and performing the query, you can see that the Id property is not loaded.

Let me know if I can do any extra clarification
